lxd/storage/backend_lxd.go

```diff
@@ -743,9 +743,19 @@ func (b *lxdBackend) CreateInstanceFromBackup(srcBackup backup.Info, srcData io.
 		})
 	}
 
-	// Update pool information in the backup.yaml file.
+	// Update information in the backup.yaml file.
 	err = vol.MountTask(func(mountPath string, op *operations.Operation) error {
-		return backup.UpdateInstanceConfigStoragePool(b.state.DB.Cluster, srcBackup, mountPath)
+		err := backup.UpdateInstanceConfigStoragePool(b.state.DB.Cluster, srcBackup, mountPath)
```

Comment:
Can we update and rename UpdateInstanceConfigStoragePool to just `UpdateInstanceConfig` so that it does both storage pool and instance name update. This would avoid the need to open, parse, encode, and write the file twice.

Also feel free to change UpdateInstanceConfigStoragePool to not use that unusual `f` function, as it seems to have been copied over from something other area of the code in the past and seems unnecessary in its current place.